File: android-core/src/main/java/com/mparticle/internal/listeners/GraphListener.java
package com.mparticle.internal.listeners;

import android.os.Message;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

interface GraphListener {
    void onCompositeObjects(@Nullable Object child, @Nullable Object parent);
    void onThreadMessage(@NonNull String handlerName, @Nullable Message msg, boolean onNewThread, @Nullable StackTraceElement[] stackTrace);
}

File: android-core/src/main/kotlin/com/mparticle/internal/listeners/GraphListener.kt
package com.mparticle.internal.listeners

import android.os.Message

interface GraphListener {
    fun onCompositeObjects(child: Any?, parent: Any?)

    fun onThreadMessage(
        handlerName: String,
        msg: Message?,
        onNewThread: Boolean,
        stackTrace: Array<StackTraceElement?>?
    )
}
